waspc/src/Wasp/Node/Version.hs

```diff
@@ -16,26 +13,53 @@ import qualified Text.Regex.TDFA as R
 import qualified Wasp.SemanticVersion as SV
 import Wasp.Util (indent)
 
+-- | Wasp supports any node version equal or greater to this version.
+-- | We usually keep this one equal to the latest LTS.
+oldestWaspSupportedNodeVersion :: SV.Version
+oldestWaspSupportedNodeVersion = SV.Version 20 8 1
+
+data VersionCheckResult
+  = VersionCheckFail !ErrorMessage
+  | VersionCheckSuccess !(Maybe WarningMessage) !SV.Version
+
+type WarningMessage = String
+
+type ErrorMessage = String
+
 -- | Gets the installed node version, if any is installed, and checks that it
 -- meets Wasp's version requirement.
---
--- Returns a string representing the error
--- condition if node's version could not be found or if the version does not
--- meet the requirements.
-getAndCheckNodeVersion :: IO (Either String SV.Version)
+getAndCheckNodeVersion :: IO VersionCheckResult
 getAndCheckNodeVersion =
   getNodeVersion >>= \case
-    Left errorMsg -> return $ Left errorMsg
-    Right nodeVersion ->
-      if SV.isVersionInRange nodeVersion nodeVersionRange
-        then return $ Right nodeVersion
-        else return $ Left $ makeNodeVersionMismatchMessage nodeVersion
+    Left errorMsg -> return $ VersionCheckFail errorMsg
+    Right userNodeVersion ->
+      return $
+        if SV.isVersionInRange userNodeVersion $ SV.Range [SV.gte oldestWaspSupportedNodeVersion]
+          then VersionCheckSuccess Nothing userNodeVersion
```

Comment:
Is there are a case where you are using the `nodeVersion` in the success scenario?

Reply:
You mean where I have a warning message in the success scenario? Something like `VersionCheckSuccess (Maybe "oh oh watch out") userNodeVersion`?

Currently there is no such usage in the code, but idea was to produce a warning if you are using an odd version of Node, I left that for later though but wanted to have some space for it when we will need it. Hm maybe I overengineered, but I think it is ok, it is a small thing.

Reply:
I think he meant "Why does the `VersionCheckSuccess` data constructor contain the `nodeVersion` field?"

It isn't used anywhere, nor is the `WarningMessage`. I know you addressed this in the other comment, but I think we should remove both of those.

We can add them to the data constructor when they become necessary. Dragging them around the codebase and always ignoring them IMO adds noise, and I found it confusing.

Reply:
I removed them, as it is true that they are not used anywhere.

I do wonder though @sodic why did you find it confusing? I thought code is pretty clear -> in case of success, you get user's node version and possibly a warning if any.

Reply:
It's confusing that we have them and never use them, that's what I meant. Their names are fine, but seeing them defined and always ignored just leaves me wondering what I am missing.